docs/dev/datastream_api.md

```diff
@@ -497,7 +497,9 @@ env.generateSequence(1,10).map(new MyMapper()).setBufferTimeout(timeoutMillis);
 LocalStreamEnvironment env = StreamExecutionEnvironment.createLocalEnvironment
 env.setBufferTimeout(timeoutMillis)
 
-env.genereateSequence(1,10).map(myMap).setBufferTimeout(timeoutMillis)
+def myMap : Long => Long = _ + 1 // `generateSequence` returns a `DataStream[Long]` type
```

Comment:
This change makes the java/scala documentation inconsistent. The fact that generateSequence returns longs is also irrelevant to this section as it is about setting the buffer timeout.

Reply:
I'm taking the perspective of the absolute beginner since this doc was linked from the QuickStart section; when i started with the QuickStart section most parts of the doc were good enough (imo) to guide me w/o referring to the javadocs (which i think makes perfect sense) so i thought it might be a good idea to continue this tradition and make things explicitly clear. What do you think @zentol ?